projects/packages/analyzer/scripts/jp-analyze-parallel.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+ini_set( 'memory_limit', '512M' );
+
+require dirname( __DIR__ ) . '/vendor/autoload.php';
+
+use Automattic\Jetpack\Analyzer\Locker;
+use Automattic\Jetpack\Analyzer\Model;
+
+$slurper_path = dirname( __DIR__ ) . '/slurper/plugins';
+
+$jetpack_exclude = array( '.git', 'vendor', 'tests', 'docker', 'bin', 'scss', 'images', 'docs', 'languages', 'node_modules' );
+// $jetpack_new_path = '/Users/brbrr/Developer/a8c/jetpack/projects/plugins/jetpack';
+$jetpack_new_path = '/var/www/html/wp-content/plugins/jetpack';
+$jetpack_old_path = $slurper_path . '/jetpack-production';
+
+class ScanManager {
+	private $count          = 0;
+	private $procs          = array();
+	private $pipez          = array();
+	private $descriptorspec = array(
+		0 => array( 'pipe', 'r' ),
+		1 => array( 'pipe', 'w' ),
+		2 => array( 'file', '/tmp/error-output.txt', 'a' ),
```

Comment:
My idea was to write the error output into a file. Is it something wrong with this approach?

Reply:
Now that I've looked through the whole PR, I think "if the use case for this script is narrow enough" is indeed the case. The script is only intended to be run as a background process from the API endpoint, and not by anyone manually, so writing to a hard-coded output file is probably ok.

On the other hand, this might be dangerous on a shared system. What happens if someone creates this file as a symlink to some file writable by the webserver user, will PHP follow the symlink and overwrite some file somewhere?

Reply:
My understanding is that apache's `PrivateTmp` should take care of the issue you mentioned.